opentelemetry-sdk/tests/logs/test_export.py

```diff
@@ -463,161 +468,129 @@ def test_validation_negative_max_queue_size(self):
+            # Should not reach this during the test, instead export should be called when delay millis is hit.
+            max_queue_size=15,
+            max_export_batch_size=15,
+            schedule_delay_millis=100,
+        )
+        for _ in range(15):
+            log_record_processor.emit(EMPTY_LOG)
+            time.sleep(0.11)
+        exporter.export.assert_has_calls(
+            [call([EMPTY_LOG]) for _ in range(15)]
+        )
 
+    def test_logs_flushed_before_shutdown_and_dropped_after_shutdown(self):
+        exporter = Mock()
+        log_record_processor = BatchLogRecordProcessor(
+            exporter=exporter,
+            # Neither of these thresholds should be hit before test ends.
+            max_queue_size=15,
+            max_export_batch_size=15,
+            schedule_delay_millis=30000,
+        )
+        # This log should be flushed because it was written before shutdown.
+        log_record_processor.emit(EMPTY_LOG)
         log_record_processor.shutdown()
         self.assertTrue(exporter._stopped)
 
-        finished_logs = exporter.get_finished_logs()
-        expected = [
-            ("warning message: possible upcoming heatwave", "WARN"),
-            ("Very high rise in temperatures across the globe", "ERROR"),
-            (
-                "Temperature hits high 420 C in Hyderabad",
-                "CRITICAL",
-            ),
-        ]
-        emitted = [
-            (item.log_record.body, item.log_record.severity_text)
-            for item in finished_logs
-        ]
-        self.assertEqual(expected, emitted)
-        for item in finished_logs:
-            self.assertEqual(item.instrumentation_scope.name, "shutdown")
-
-    def test_force_flush(self):
-        exporter = InMemoryLogExporter()
-        log_record_processor = BatchLogRecordProcessor(exporter)
-
-        provider = LoggerProvider()
-        provider.add_log_record_processor(log_record_processor)
-
-        logger = logging.getLogger("force_flush")
-        logger.propagate = False
-        logger.addHandler(LoggingHandler(logger_provider=provider))
-
-        logger.critical("Earth is burning")
-        log_record_processor.force_flush()
-        finished_logs = exporter.get_finished_logs()
-        self.assertEqual(len(finished_logs), 1)
-        log_record = finished_logs[0].log_record
-        self.assertEqual(log_record.body, "Earth is burning")
-        self.assertEqual(log_record.severity_number, SeverityNumber.FATAL)
-        self.assertEqual(
-            finished_logs[0].instrumentation_scope.name, "force_flush"
+        with self.assertLogs(level="WARNING") as log:
+            # This log should not be flushed.
+            log_record_processor.emit(EMPTY_LOG)
+            self.assertEqual(len(log.output), 1)
+            self.assertEqual(len(log.records), 1)
+            self.assertIn("Shutdown called, ignoring log.", log.output[0])
+        exporter.export.assert_called_once_with([EMPTY_LOG])
```

Comment:
This might be easier to read if yuo move this to right after the shutdown() call on L518